src/main.rs

```diff
@@ -51,19 +47,16 @@ async fn main() -> Result<()> {
 
     // Setup the eth provider
     let starknet_provider = Arc::new(starknet_provider);
-    let eth_provider = EthDataProvider::new(db.clone(), starknet_provider.clone()).await?;
-    let alchemy_provider = AlchemyDataProvider::new(eth_provider.clone());
-    let pool_provider = PoolDataProvider::new(eth_provider.clone());
-    let debug_provider = DebugDataProvider::new(eth_provider.clone());
+
+    let eth_client = EthClient::try_new(starknet_provider, db.clone()).await.expect("failed to start ethereum client");
+    let eth_provider = eth_client.eth_provider();
 
     // Setup the retry handler
-    let retry_handler = RetryHandler::new(eth_provider.clone(), db);
+    let retry_handler = RetryHandler::new(eth_provider, db);
     retry_handler.start(&tokio::runtime::Handle::current());
```

Comment:
Here we are missing the tricky part where we start the account manager service with the mempool no?

Should be something like `account_manager.start(&eth_data_provider)`. The problem is that here we need to share a shared pointer between the eth client and the account manager service in order to fill on one side (via the send_raw_transaction) and empty on the other side (via the account manager service) the same mempool. The last time I've checked that I had a borrow checker issue between the two